packages/run-protocol/src/runStake/attestation.js

```diff
@@ -0,0 +1,254 @@
+    const s = await E(lienBridge).getAccountState(address, stakeBrand);
+    const unlocked = subtractOrZero(s.total, s.locked);
+    checkDelta(toLien, 'unliened', s.total, s.liened);
+    checkDelta(toLien, 'bonded and unliened', s.bonded, s.liened);
+    checkDelta(toLien, 'unlocked and unliened', unlocked, s.liened);
+  };
+
+  /** @type {Store<Address, Amount<'nat'>>} */
+  const amountByAddress = makeStore('amount');
+
+  /**
+   * @param { string } address
+   * @param { Amount<'nat'> } lienDelta
+   */
+  const mintAttestation = async (address, lienDelta) => {
+    assert.typeof(address, 'string');
+    // This account state check is primarily to provide useful diagnostics.
+    // Since things like the bonded amount can change out from under us,
+    // we shouldn't rely on it completely.
+    // TODO: consider performance implications vs. diagnostics.
+    await assertAccountStateOK(address, lienDelta);
+
+    const current = getOrElse(amountByAddress, address, () => empty);
+    const target = add(current, lienDelta);
+
+    // Since our mint is not subject to normal supply/demand,
+    // minting here and failing in setLiened below does no harm.
+    const attestationPaymentP = mintZCFMintPayment(
+      zcf,
+      zcfMint,
+      wrapLienedAmount(address, lienDelta),
+    );
+
+    // If setLiened returns successfully, then
+    // the preconditions were indeed met and we
+    // can update the stored liened amount.
+    await E(lienBridge).setLiened(address, current, target);
+    amountByAddress.set(address, target);
+    return attestationPaymentP;
+  };
+
+  /** @param {ZCFSeat} seat */
+  const returnAttestation = async seat => {
+    assertProposalShape(seat, { give: { Attestation: null }, want: {} });
+
+    /** @type {Amount<'copyBag'>} */
+    const attestationAmount = seat.getAmountAllocated(KW.Attestation);
+    const { address, lienedAmount: amountReturned } =
+      unwrapLienedAmount(attestationAmount);
+
+    const liened = amountByAddress.get(address); // or throw
+    const updated = subtract(liened, amountReturned); // or throw
+
+    await E(lienBridge).setLiened(address, liened, updated);
+    // COMMIT. Burn the escrowed attestation payment and exit the seat
+    amountByAddress.set(address, updated);
+    zcfMint.burnLosses(harden({ Attestation: attestationAmount }), seat);
+    seat.exit();
+  };
+
+  /**
+   * Get liened amount synchronously.
+   *
+   * @param {string} address
+   * @param {Brand<'nat'>} brand
+   */
+  const getLiened = (address, brand) => {
+    assert.typeof(address, 'string');
+    assert.equal(
+      brand,
+      stakeBrand,
+      X`This contract can only make attestations for ${brand}`,
+    );
+    return getOrElse(amountByAddress, address, () => empty);
+  };
+
+  const lienMint = harden({
+    mintAttestation,
+    returnAttestation,
+    getLiened,
+    /**
+     * @param { Amount<'copyBag'> } attAmt
+     * @throws { Error } if `attAmt` payload length is not 1
```

Comment:
the type being thrown is not informative here. consider,
```suggestion
     * @throws if `attAmt` payload length is not 1
```